### PR TITLE
Consume optimization fuel from the MIR inliner

### PR DESCRIPTION
**Pull request by wesleywiser**
This makes it easier to debug mis-optimizations that occur during
inlining. Thanks to @nikomatsakis for the suggestion!

### CODE REVIEW

File: src/librustc_mir/transform/inline.rs [hunk 1]
```diff
@@ -198,6 +198,16 @@ impl<'a, 'tcx> Inliner<'a, 'tcx> {
...
     }
 
+    fn consider_optimizing(&self,
+                           callsite: CallSite<'tcx>,
+                           callee_mir: &Mir<'tcx>)
+                           -> bool
+    {
+        debug!("consider_optimizing({:?})", callsite);
+        self.tcx.consider_optimizing(|| format!("Inline {:?} into {:?}", callee_mir.span, callsite))
+            && self.should_inline(callsite, callee_mir)
```
Comment (eddyb) [inline src/librustc_mir/transform/inline.rs]: I think the order here is wrong, IMO we should first check if we can inline, that would reduce a lot of fuel uses and make it easier to track down a specific misinline.

Comment (nikomatsakis) [reply src/librustc_mir/transform/inline.rs]: that makes sense to me

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
r=me with the order switched

File: src/librustc_mir/transform/inline.rs (hunk 1 above)
Comment (wesleywiser) [reply src/librustc_mir/transform/inline.rs]: Sure, I can do.

**Pull request merged**